cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java

```diff
@@ -756,6 +764,30 @@ private String pgId(CQLTermNode node) throws QueryValidationException {
     }
   }
 
+  private String arrayNode(String index, CQLTermNode node, org.folio.rest.persist.ddlgen.Modifier relationModifier,
+    String modifierValue, DbIndex dbIndex, CqlModifiers modifiers) throws QueryValidationException {
+
+    StringBuilder res = new StringBuilder();
+
+    IndexTextAndJsonValues vals = new IndexTextAndJsonValues();
+    vals.setIndexText(index2sqlText("t.c", relationModifier.getSubfield()));
+
+    final String table = this.jsonField.split("\\.")[0];
+    final String field = this.jsonField.split("\\.")[1];
+
+    res.append("id in (select t.id from (select id as id, jsonb_array_elements(");
+    res.append(field + "->'" + index + "') as c from " + table + ") as t where t.c");
+    res.append(" @> '{\"");
+    // pass the name from the schema rather than CQL because case is lost
+    res.append(Cql2SqlUtil.cql2string(relationModifier.getModifierName()));
```

Comment:
This is a JSON string embedded in an SQL string. For the JSON string " needs to be masked.
We propably need Cql2SqlUtil.cql2JsonSqlString(String)

Reply:
Not necessary. It's a quoted string so while one can make bad JSON it's still quoted (with single quotes) and it's therefore impossible to unquote that .

Reply:
or create

    @> json_build_object('someName', 'sameValue')

using

    res.append(" @> json_build_object('");
    res.append(Cql2SqlUtil.cql2string(relationModifier.getModifierName()));
    res.append("', '");
    res.append(Cql2SqlUtil.cql2string(modifierValue));

Reply:
Tried it, but couldn't get it to work.